MAHS/Assets/SimpleEnv/Scripts/TrainingAreaScript.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+using Unity.MLAgents;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class TrainingAreaScript : MonoBehaviour
+{
+    [HideInInspector]
+    public bool isAnyHiderSeen;
+    [HideInInspector]
+    public List<AgentScript> agents;
+    void Awake()
+    {
+        Academy.Instance.OnEnvironmentReset += ResetEnv;
+        Academy.Instance.AgentPreStep += PreStep;
+    }
+
+    public int episodeLength = 1000;
+    public int preparingPhaseLength = 400;
+    
+    public Transform box1;
+    public Transform box2;
+    public void ResetEnv()
+    {
+        box1.transform.localPosition = new Vector3(Random.value * 4.25f + 0.25f, 0.5f, -Random.value * 4.25f - 0.25f);
+        box2.transform.localPosition = new Vector3(Random.value * 4.25f + 0.25f, 0.5f, -Random.value * 4.25f - 0.25f);
+    }
+
+    void PreStep(int i)
+    {
+        float hidersReward = 1.0f;
```

Comment:
Let's simplify it as following:
`float hidersReward = isAnyHiderSeen ? -1.0f : 1.0f;`